ODIN_II/SRC/include/HardSoftLogicMixer.hpp

```diff
@@ -0,0 +1,107 @@
+/*
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following
+ * conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+ * OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+ * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifndef HARD_SOFT_LOGIC_MIXER_HPP
+#define HARD_SOFT_LOGIC_MIXER_HPP
+
+#include "odin_types.h" // netlist_t, config_t
+#include "MixingOptimization.hpp"
+
+class HardSoftLogicMixer {
+  public:
+    HardSoftLogicMixer(const config_t& configuration);
+
+    /* Returns whether the hard blocks and soft logic implementation 
+     * of multipliers should be mixed in the process of synthesis
+     */
+    bool softenable(operation_list type);
```

Comment:
pass in the node here rather and inside the function go access it.

Reply:
Done, thanks for the comment!

Reply:
Done, thanks for the comment!